### ISSUE TITLE
[Bug] Crystalite Armor not giving armor points and Crystalite elytra knockback resistance bugged

### ISSUE THREAD

**Issue by vvollpp** [quiqueck/BetterEnd#457]
### What happened?

_(I can't select Minecraft 1.21.1 under the "Versions" for this bug report, so I used 1.20.4, the most recent one after that)_

I've noticed that the Crystalite armor doesn't give any armor points, only the chestplate / elytra does. And I've just found out why. The chestplate says "When on Body:" where "Body" means armor for the chest/chestplates, same for the elytra. For boots it should be "When on Feet", for leggins "When on Legs", helmet "When on Head".
The problem is, for me, it says "When on Body" for all Crystalite armor pieces, and apparently that means they don't give any of the armor points, even though you can still equip them. Only the chestplate gives you the proper armor points, since that's the only armor piece that's supposed to be "on Body". You still get all the debuffs when wearing the whole set, though.
I've looked at videos and this seems to be a new bug in this particular version. Here's an example when he hovers the mouse over the bookmark of REI (from https://youtu.be/gxOTXlnXwqU?si=pV93uyzjzpKFhGJx&t=1215) **(The video is 2 years old)**
![image](https://github.com/user-attachments/assets/87f58f77-57a3-4b01-b66e-a02ec8d827e3)

Full armor set equipped:
![image](https://github.com/user-attachments/assets/475f3b6c-5723-4899-8b0a-d6139e2b182f)
Yet it only gives the armor points for the chestplate:
![image](https://github.com/user-attachments/assets/a3022797-36d4-437c-ab72-ec91f4da70ac)

I've compared it to the Fireruby armor set as an example, since that one works just fine:
Fireruby Boots show the stats as they should be:
![image](https://github.com/user-attachments/assets/0489826a-9e11-4d57-9b9e-39d9c7eb0122)
Crystalite Boots are set to "When on Body" and therefor don't give any armor points (Swiftness still works):
![image](https://github.com/user-attachments/assets/0401f6e1-faf5-4bd9-8a7e-add503bca75c)
Fireruby Chestplate as it should be:
![image](https://github.com/user-attachments/assets/1274f830-6efd-4c37-bc76-9dd5c1c7ed8c)
Crystalite Chestplate, the only Crystalite armor piece that works:
![image](https://github.com/user-attachments/assets/62fcaee2-0034-4959-a436-77377634ee4a)
(This is also the case for the leggings and helmet)

### ALSO ANOTHER SIGNIFICANT BUG: (at least I think it's a bug)
While testing around with the Crystalite armor bug, I've noticed that the Crystalite elytra's stat card says "When in Main Hand: +5 Knockback Resistance" which doesn't really make sense. You get knockback resistance when holding it in your hand, but not when wearing it? Surely that can't be right. It does still give the +6.4 armor and +0.96 armor toughness.
![image](https://github.com/user-attachments/assets/a362f838-36cb-4950-b9bb-7c4bc6028d83)

And I'm just now seeing that only the elytra gets knockback resistance. And none of the Crystalite armor pieces get the "+ Max Health" spec either. Here's the screenshot from the video again.
![image](https://github.com/user-attachments/assets/f38a9958-39cb-4009-8a66-673668747f08)
Compared to the leggings in my current version:
![image](https://github.com/user-attachments/assets/3c85f4f4-2f21-4d77-925d-955f9e44cbe4)

It says "When on Head"... On a leggings... Needless to say you can't put the leggings on your head and thus the Max Health effect is not possible to get.
The Crystalite leggings is also the only Crystalite armor piece that has any kind of mention of more max health in it's stats. The others just don't have any max health additions at all.

And just for completion's sake, here's the screenshot of the Crystalite helmet, which only gives you the Blindness Resistance effect (since that is the only stat that's under the category "When on Head"):
![image](https://github.com/user-attachments/assets/2f6e5721-1464-466c-b041-cdda63f247e6)

This bug report has gotten way more in depth than I expected it to. Hopefully this will be helpful enough, though. If you need any more information, please let me know, I'd gladly help further. Also, if this post seems too cluttered or not informative enough, I apologize, I'm not used to writing them.
Thanks for reading through this. Thanks for making these awesome mods.

### BetterEnd

21.0.8

### BCLib

21.0.12

### Fabric API

0.102.0+1.21

### Fabric Loader

0.16.0

### Minecraft

1.20.4

### Other Mods

```shell
I have a lot of mods installed, 60 in total, including all libraries, etc. I don't think any of them should cause any problems, but I'll list them alphabetically just in case:
AppleSkin
Architectury API
Axes Are Weapons
Balm
BCLip
Better Advancements
Better End
Better Nether
Biomes O' Plenty
Blur+
Carry On
cat_jam
Client Tweaks
Cloth Config API
Clumps
Cobweb [an API]
Collective [a library]
Configured
Craft Slabs Back into Blocks
Creeper Overhaul
Duckling
Easy Anvils
Fabric API
Fabric Language Kotlin
Forge Config API Port
GeckoLib
GlitchCore [a library]
Grass Overhaul
Harvest with ease
Inventory Totem
Iris Shader
Jade
Just Enough Items (JEI) [I have this disabled since it doesn't fully work with Better End, I use REI instead)
KleeSlabs
libIPN
M.R.U [a library]
Mo Glass
Mod Menu
Model Gap Fix
Mouse Tweaks
Necronomicon API
Obsidian UI [an API]
Perfect Accuracy
Puzzles Lib
Resourceful Config
Resourceful Lib
Roughly Enough Items (REI)
RyoamicLights
Snow Under Trees
Sodium
Spyglass Improvements
Structory
TerraBlender [a library]
Text Placeholder API
TipTheScale
Tree Harvester
Vein Mining
WorldWeaver [API for Better End I believe]
YetAnotherConfigLib
Zoomify
```

**Comment by vvollpp** [comment]
Again, if something's unclear or you need more info, tell me and I'll try my best to help

**Comment by OpenBagTwo** [comment]
This is a duplicate of #446, and a fix is awaiting review in #437.

Given that this has been reported multiple times, I'd suggest closing this issue.

**Comment by vvollpp** [comment]
Oh, okay, sorry, I should've looked if it was mentioned already. My bad.
Good luck fixing this issue, I'll be looking forward to the fix

**Comment by vvollpp** [comment]
It says to use the Fabric ArmorRenderer in order for it to work now. However, I can't that anywhere. Neither in the curseforge mod search for all mods, nor in the search for 1.21 fabric mods.
Where can I find ArmorRenderer?

**Comment by OpenBagTwo** [comment]
That's should just be part of the Fabric API mod.

**Comment by vvollpp** [comment]
Still not completely fixed, though.
![image](https://github.com/user-attachments/assets/9cda0c03-d067-45be-b2ae-bb6184fa8a38)
![image](https://github.com/user-attachments/assets/6e42f1a8-0640-476a-a410-356e79bbab0b)

The rest seems fine, as long as nothing else besides the leggings is supposed to have extra health
![image](https://github.com/user-attachments/assets/a91a32a2-56df-44c3-9d02-2e76543ed011)
![image](https://github.com/user-attachments/assets/31109348-d706-4abe-b60e-41b6077cbc39)
![image](https://github.com/user-attachments/assets/c6fc8dca-fc2b-4b4c-8d7b-007e2dea8e36)

**Comment by vvollpp** [comment]
Nvm. The armor and armor toughness stats still don't match the ones they're [supposed to have](https://github.com/quiqueck/BetterEnd/pull/437#issuecomment-2241797851)

**Comment by OpenBagTwo** [comment]
@vvollpp what version / which build of BetterEnd are you running? Again, this fix has not yet been merged.

**Comment by vvollpp** [comment]
The newest one that came out, 20.0.9

**Comment by OpenBagTwo** [comment]
As I said, that PR still has not been merged, so the fix is not present in any release. If you'd like to try out the changes introduced in that PR, I made [an unofficial build available,](https://github.com/OpenBagTwo/BetterEnd/releases/tag/21.0.10%2Bopenbagtwo.240813) but use it at your own risk, and don't report any problems you find with that build to Quiqueck's repo unless you've verified that the issues are also present on the most recent official release.

**Comment by vvollpp** [comment]
I honestly don't know half of what all that means. I thought the new update yesterday for BetterEnd was supposed to fix it. I guess I just got confused. I'll just be waiting for the official fix, then

**Comment by OpenBagTwo** [comment]
> I honestly don't know half of what all that means.

Sorry—I forget sometimes that not everyone who signs up for a GitHub account is super technical.

I'll break it down:

1. The issue is known (reported in issue #446)
2. A fix has been proposed (pull request #437), but
3. The mod's maintainer has yet to approve the fix and incorporate it ("merge" it) into the official codebase
4. Which is why it's not part of any official update.
5. I have made my own _unofficial_ update for people to try out, but
6. 
   > I'll just be waiting for the official fix, then
   
   is absolutely the action I would recommend.

**Comment by vvollpp** [comment]
Ah alright, got it. Thanks a lot for breaking it down. I thought you were the dev behind BetterEnd etc. so I just got more confused.

Yeah, I created a GitHub account just for this bug report and a suggestion I made for JEI. Never really used it before otherwise either.

Again, thanks for being so understanding and explain it more clearly. Appreciate it
